build/index.js

```diff
@@ -35,124 +42,109 @@ const bootstrap = () => {
-    const extractionPath = `${tmpPath}/icons`;
-    zip.extractAllTo(extractionPath, true);
-    return Promise.resolve(`${extractionPath}/${basename(archive)}`);
-  } catch (e) {
-    return Promise.reject(`Could not extract icons: ${e}`);
-  }
+const createIconName = (name) => {
+  return require("camelcase")(name);
 };
 
-const createModuleName = (name) => {
-  return require("pascal-case").pascalCase(name);
+const createElmContent = ({ iconName, body, funcNames }) => {
+  return funcNames.reduce((acc, fn) => {
+    const ret = acc.replace(
+      new RegExp(`${fn}(:| )`, "gm"),
+      `${fn.replace("view", iconName)}$1`
+    );
+    return ret;
+  }, body);
 };
 
-const createElmContent = ({ moduleName, imports, exports, body }) => {
-  const moduleExposing =
-    exports.kind === "SOME" ? `${exports.entries.join(", ")}` : "..";
-  const moduleImports = imports.map(({ module, exposing }) => {
-    return `import ${module} exposing (${
-      exposing.kind === "SOME" ? `${exposing.entries.join(", ")}` : ".."
-    })`;
-  });
-  return `
-module ${moduleName} exposing (${moduleExposing})
-
-${moduleImports.join("\n")}
-
-${body}
-    `;
-};
+const convertIcon = (iconsPath) => async (icon) => {
+  const iconName = createIconName(basename(icon));
 
-const convertIcon = (iconsPath) => (icon) => {
-  const pathToIcon = `${iconsPath}/${icon}`;
-  const moduleName = createModuleName(basename(icon));
-  const elmModulePath = `${elmPath}/${moduleName}.elm`;
-  const convertor = require("svg-to-elm").default;
+  try {
+    const pathToIcon = `${iconsPath}/${icon}`;
+    const convertor = require("svg-to-elm").default;
+    const res = await convertor.parse(pathToIcon, { moduleName: iconName });
 
-  convertor.parse(pathToIcon, { moduleName: elmModulePath }).then((res) => {
     if (!res.success) {
       throw new Error(res.message);
     }
 
-    try {
-      const elmContent = createElmContent({
-        moduleName: `ZondIcons.${moduleName}`,
-        imports: res.imports,
-        body: res.viewBody,
-        exports: res.moduleExposing,
-      });
-
-      fs.writeFileSync(elmModulePath, elmContent);
-    } catch (e) {
-      throw new Error(`Could not write Elm (${elmModulePath}): ${e}`);
-    }
-  });
+    return createElmContent({
+      iconName,
+      body: res.viewBody,
+      funcNames: res.moduleExposing.entries || [],
+    });
+  } catch (e) {
+    throw new Error(`Could not convert ${icon}: ${e}`);
+  }
 };
+
 const isSVGIcon = (file) => path.extname(file).toLowerCase() === ".svg";
 
-const convertIcons = (iconsPath) => {
+const convertIcons = async (iconsPath) => {
   try {
     const convert = convertIcon(iconsPath);
 
-    fs.readdirSync(iconsPath)
-      .filter(isSVGIcon)
-      .forEach(convert);
-    return Promise.resolve();
+    const buffer = await Promise.all(
+      fs.readdirSync(iconsPath).reduce((acc, file) => {
+        if (isSVGIcon(file)) {
+          return [...acc, convert(file)];
+        }
+        return acc;
```

Comment:
this is the perfect place for a ternary IMO

Reply:
Will refactor

Reply:
Done